### ISSUE TITLE
qiime2 import error due to multiple users needing to access /tmp/qiime2 on Ubuntu server

### ISSUE THREAD

**Issue by jcmcnch** [qiime2/qiime2#639]
**Bug Description**

I ran into a cryptic permissions error when trying out the same import command in two different user accounts on an Ubuntu remote server. Both are using the same `qiime2-2022.8` deployment, installed in the same way (according to qiime2 docs for conda, only difference being I used mamba to speed things up). 

Import command:

```
qiime tools import \
  --type 'SampleData[PairedEndSequencesWithQuality]' \
  --input-path manifest.tsv \
  --output-path 16s.qza \
  --input-format PairedEndFastqManifestPhred33V2
```

This is what I was seeing in terms of an error message:

```
Traceback (most recent call last):
  File "/home/inigo/miniconda3/envs/qiime2-2022.8/lib/python3.8/site-packages/q2cli/builtin/tools.py", line 157, in import_data
    artifact = qiime2.sdk.Artifact.import_data(type, input_path,
  File "/home/inigo/miniconda3/envs/qiime2-2022.8/lib/python3.8/site-packages/qiime2/sdk/result.py", line 299, in import_data
    return cls._from_view(type_, view, view_type, provenance_capture,
  File "/home/inigo/miniconda3/envs/qiime2-2022.8/lib/python3.8/site-packages/qiime2/sdk/result.py", line 334, in _from_view
    artifact._archiver = archive.Archiver.from_data(
  File "/home/inigo/miniconda3/envs/qiime2-2022.8/lib/python3.8/site-packages/qiime2/core/archive/archiver.py", line 379, in from_data
    path, process_alias, cache = cls._make_temp_path(uuid)
  File "/home/inigo/miniconda3/envs/qiime2-2022.8/lib/python3.8/site-packages/qiime2/core/archive/archiver.py", line 274, in _make_temp_path
    cache = get_cache()
  File "/home/inigo/miniconda3/envs/qiime2-2022.8/lib/python3.8/site-packages/qiime2/core/cache.py", line 67, in get_cache
    _CACHE.temp_cache = Cache()
  File "/home/inigo/miniconda3/envs/qiime2-2022.8/lib/python3.8/site-packages/qiime2/core/cache.py", line 169, in __init__
    self.path = pathlib.Path(self._get_temp_path())
  File "/home/inigo/miniconda3/envs/qiime2-2022.8/lib/python3.8/site-packages/qiime2/core/cache.py", line 264, in _get_temp_path
    os.chmod(cache_dir, sticky_permissions)
PermissionError: [Errno 1] Operation not permitted: '/tmp/qiime2'

An unexpected error has occurred:

  [Errno 1] Operation not permitted: '/tmp/qiime2'

See above for debug info.
```

**Steps to reproduce the behavior**
1. Run import command for user 1. Import succeeds.
2. Run import command for user 2. Import fails with output above.
3. Return to user 1 terminal session. Run `rm -rf /tmp/qiime2`
4. Return to user 2. Retry import. Import succeeds.

I tried changing permissions from user 1's account (`chmod -R 777 /tmp/qiime2`) but it didn't fix the problem. Only solution I've come up with so far is the `rm -rf /tmp/qiime2`

I also tried  `qiime2-2022.2` with the failing user, and it succeeded. So it seems to be something specific to `qiime2-2022.8` as far as I can tell. So I guess I'll just use the previous version until this is fixed or someone can tell me what I'm doing wrong.

**Expected behavior**
I expected both user sessions to be able to write whatever tmp files they need without interfering with one another.

**Computation Environment**
- Ubuntu 20.04.4 LTS
- qiime2-2022.8

**Comment by ebolyen** [comment]
Hi @jcmcnch, sorry for the issues, this is a new feature and clearly isn't working correctly in your situation. Could you tell us more about the filesystem on /tmp/ and run `ls -al` on `/tmp/qiime2`?

What should have happened is that directory should have a sticky bit set just like tmp, to allow multiple users to manage their own subdirectory without collision. This was something we were pretty sure was working on our end, but there must be some difference or subtlety to it that we've overlooked.

Also, if you don't mind what are the UIDs (just the number, we don't need the real names I think) of the two user accounts? Is one of them root?

@Oddant1 can you try to replicate this situation ASAP? @lizgehret we may need a patch release to fix this as soon as we figure out exactly where it's going wrong.

**Comment by ebolyen** [comment]
(Also, thank you for the very nicely written bug report :-) )

**Comment by Oddant1** [comment]
Hi @jcmcnch. We were able to recreate the bug locally, and it should be a fairly easy fix! It's not entirely clear how it slipped past the multi-user testing we did pre-release, but it likely had to do with the configuration of the machine we did the multi-user testing on. For now, I would advise using qiime2-2022.2 on your multi-user system until we can patch this, and I apologize for the inconvenience. If downgrading is not possible, I would advise running that `rm -rf /tmp/qiime2` after running QIIME 2 commands. Thanks for bearing with us on this, we're working on some pretty big internal changes to how QIIME 2 stores data, and it was all but inevitable that something would slip through the cracks and a patch would be needed!

**Comment by jcmcnch** [comment]
Hi @ebolyen - thanks for the prompt response and you're welcome about the bug report. It was pretty easy to track down once I figured out what was going on. Happy to help the qiime2 crew such as I'm able. I guess you don't need the info now that @Oddant1 has been able to recreate the bug, but happy to provide more info if it would be helpful. @Oddant1 thanks for the advice - I'll downgrade to 2022.2 as you suggest which is not a big deal given the last time I installed qiime I was using 2019.4... so this is already a big upgrade!
